packages/release-tool/src/index.ts

```diff
@@ -147,168 +47,296 @@ interface ExtendedGithubPrData extends Omit<GithubPrData, "mergedAt"> {
+    default: [],
+    choices: prs.map(pr => ({
+      checked: true,
+      key: pr.number,
+      name: `#${pr.number}: ${pr.title} (https://github.com/lensapp/lens/pull/${pr.number})`,
+      value: pr.number,
+      short: `#${pr.number}`,
+    })),
+    loop: false,
   });
+
+  return prs.filter(pr => answers.commits.includes(pr.number));
 }
 
-async function handleRelaventPr(pr: ExtendedGithubPrData) {
-  if (options["check-commits"] && !(await askQuestion(`Would you like to use #${pr.number}: ${pr.title}? - Y/N`))) {
-    return;
+function formatChangelog(previousReleasedVersion: string, prs: ExtendedGithubPrData[]): string {
+  const enhancementPrLines: string[] = [];
+  const bugPrLines: string[] = [];
+  const maintenencePrLines: string[] = [];
+
+  for (const pr of prs) {
+    if (isEnhancementPr(pr)) {
+      enhancementPrLines.push(formatPrEntry(pr));
+    } else if (isBugfixPr(pr)) {
+      bugPrLines.push(formatPrEntry(pr));
+    } else {
+      maintenencePrLines.push(formatPrEntry(pr));
+    }
   }
 
-  if (prBase !== "master") {
-    try {
-      const promise = exec(`git cherry-pick ${pr.mergeCommit.oid}`);
+  if (enhancementPrLines.length > 0) {
+    enhancementPrLines.unshift("## 🚀 Features", "");
+    enhancementPrLines.push("");
+  }
 
-      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      promise.child.stdout!.pipe(process.stdout);
-      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      promise.child.stderr!.pipe(process.stderr);
+  if (bugPrLines.length > 0) {
+    bugPrLines.unshift("## 🐛 Bug Fixes", "");
+    bugPrLines.push("");
+  }
 
-      await promise;
-    } catch {
-      console.error(`Failed to cherry-pick ${pr.mergeCommit.oid}, please resolve conflicts and then press enter here:`);
-      await new Promise<void>(resolve => rl.once("line", () => resolve()));
-    }
+  if (maintenencePrLines.length > 0) {
+    maintenencePrLines.unshift("## 🧰 Maintenance", "");
+    maintenencePrLines.push("");
+  }
+
+  return [
+    `## Changes since ${previousReleasedVersion}`,
+    "",
+    ...enhancementPrLines,
+    ...bugPrLines,
+    ...maintenencePrLines,
+  ].join("\n");
+}
+
+async function cherrypickCommits(prs: ExtendedGithubPrData[]): Promise<void> {
+  const rl = createInterface(process.stdin);
+  const cherrypickCommit = cherrypickCommitWith(rl);
+
+  for (const pr of prs) {
+    await cherrypickCommit(pr.mergeCommit.oid);
   }
 
-  if (isEnhancementPr(pr)) {
-    prLines.enhancement.push(getPrEntry(pr));
-  } else if (isBugfixPr(pr)) {
-    prLines.bugfix.push(getPrEntry(pr));
-  } else {
-    prLines.maintenence.push(getPrEntry(pr));
+  rl.close();
+}
+
+async function pickRelaventPrs(prs: ExtendedGithubPrData[], isMasterBranch: boolean): Promise<ExtendedGithubPrData[]> {
+  if (isMasterBranch) {
+    return prs;
   }
+
+  let selectedPrs: ExtendedGithubPrData[];
+
+  do {
+    selectedPrs = await pickWhichPRsToUse(prs);
```

Comment:
user must now always check commits? (as if `--check-commits` is always set?)

Reply:
Yes, technically, but it is easier now because the user can just press `<enter>` to select all of them.